internal/provider/notification_rule_resource.go

```diff
@@ -120,6 +122,18 @@ func (r *notificationRuleResource) Schema(ctx context.Context, req resource.Sche
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"parent_type": schema.StringAttribute{
```

Comment:
Proposal:
Instead of adding `organization_id`, we could introduce a `parent_name` field, which could be either `organizations/...` or `projects/....`. This would allow us to deprecate `project_id` and avoid introducing yet another ID field.

The main benefit is that we avoid having pairs of fields that only make sense together.